cv_bridge/include/cv_bridge/cv_bridge.h

```diff
@@ -117,10 +117,24 @@ class CV_BRIDGE_EXPORT CvImage
    * can be: jpg, jp2, bmp, png, tif at the moment
    * support this format from opencv:
    * http://docs.opencv.org/modules/highgui/doc/reading_and_writing_images_and_video.html#Mat imread(const string& filename, int flags)
+   * params are the options for cv::imencode.
+   * Default value is empty list.
    */
   sensor_msgs::msg::CompressedImage::SharedPtr toCompressedImageMsg(
-    const Format dst_format =
-    JPG) const;
+    const Format dst_format = JPG,
+    const std::vector<int> &params=std::vector<int>()) const;
+
+  /**
+   * dst_format is compress the image to desire format.
+   * Default value is empty string that will convert to jpg format.
+   * can be: jpg, jp2, bmp, png, tif at the moment
+   * support this format from opencv:
+   * http://docs.opencv.org/modules/highgui/doc/reading_and_writing_images_and_video.html#Mat imread(const string& filename, int flags)
+   * params are the options for cv::imencode.
+   * Default value is empty list.
+   */
+  sensor_msgs::msg::CompressedImage::SharedPtr toCompressedImageMsg(
+    const std::vector<int> &params=std::vector<int>(), const Format dst_format = JPG) const;
```

Comment:
I thought of this like, may be i want only provide parameters for the cv::imencode. Otherwise the dest_format need to be explicitly specified in each call we want provide params to cv::imencode.